.pre-commit-config.yaml

```diff
@@ -31,10 +35,23 @@ repos:
         args: [--fix]
       - id: ruff-format
 
-  - repo: https://github.com/pre-commit/mirrors-prettier
-    rev: v3.1.0
+  - repo: https://github.com/pre-commit/mirrors-eslint
+    rev: v8.56.0
     hooks:
-      - id: prettier
+      - id: eslint
+        additional_dependencies:
+          - prettier@3.1.0
+          - eslint@8.57.0
+          - eslint-config-prettier@8.10.0
+          - eslint-plugin-prettier@5.1.3
+          - eslint-plugin-vue@8.0.3
+          - "@vue/cli-plugin-babel@5.0.8"
+          - "@vue/cli-plugin-eslint@5.0.8"
+          - "@vue/eslint-config-prettier@8.0.0"
+          - eslint-plugin-cypress@3.3.0
+          - "@babel/core@7.24.8"
+          - "@babel/eslint-parser@7.24.8"
+          - "@babel/plugin-transform-export-namespace-from@7.24.7"
```

Comment:
It's a bit annoying that we have to do this, but alas this is the recommended approach.

One thing we might consider in the future is writing a little `invoke` script that grabs the installed versions of these packages from `yarn.lock` and updates them (replacing the equivalent `pre-commit autoupdate` functionality). However, as these rarely change I don't think its worth it for now.